component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/input/Streaming.java

```diff
@@ -0,0 +1,221 @@
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class Streaming {
+
+    private static Supplier<LocalConfiguration> defaultLocalConfiguration = () -> new LocalConfiguration() {
+
+        @Override
+        public String get(final String key) {
+            return null;
+        }
+
+        @Override
+        public Set<String> keys() {
+            return emptySet();
+        }
+    };
+
+    public static RetryConfiguration loadRetryConfiguration(final String plugin) {
+        // note: this configuratoin could be read on the mapper too and distributed
+        final LocalConfiguration configuration = ofNullable(ContainerFinder.Instance.get().find(plugin))
+                .map(it -> it.findService(LocalConfiguration.class))
+                .orElseGet(defaultLocalConfiguration);
+        final int maxRetries = ofNullable(configuration.get("talend.input.streaming.retry.maxRetries"))
+                .map(Integer::parseInt)
+                .orElse(Integer.MAX_VALUE);
+        return new RetryConfiguration(maxRetries, getStrategy(configuration));
+    }
+
+    public static RetryStrategy getStrategy(final LocalConfiguration configuration) {
+        switch (ofNullable(configuration.get("talend.input.streaming.retry.strategy")).orElse("constant")) {
+        case "exponential":
+            return new RetryConfiguration.ExponentialBackoff(
+                    ofNullable(configuration.get("talend.input.streaming.retry.exponential.exponent"))
+                            .map(Double::parseDouble)
+                            .orElse(1.5),
+                    ofNullable(configuration.get("talend.input.streaming.retry.exponential.randomizationFactor"))
+                            .map(Double::parseDouble)
+                            .orElse(.5),
+                    ofNullable(configuration.get("talend.input.streaming.retry.exponential.maxDuration"))
+                            .map(Long::parseLong)
+                            .orElse(TimeUnit.MINUTES.toMillis(5)),
+                    ofNullable(configuration.get("talend.input.streaming.retry.exponential.initialBackOff"))
+                            .map(Long::parseLong)
+                            .orElse(TimeUnit.SECONDS.toMillis(1)),
+                    0);
+        case "constant":
+        default:
+            return new RetryConfiguration.Constant(
+                    ofNullable(configuration.get("talend.input.streaming.retry.constant.timeout"))
+                            .map(Long::parseLong)
+                            .orElse(500L));
+        }
+    }
+
+    public static StopStrategy loadStopStrategy(final String plugin) {
+        final LocalConfiguration configuration = ofNullable(ContainerFinder.Instance.get().find(plugin))
+                .map(it -> it.findService(LocalConfiguration.class))
+                .orElseGet(defaultLocalConfiguration);
+        final Long maxReadRecords = ofNullable(configuration.get("talend.input.streaming.maxRecords"))
```

Comment:
Can I configure those configuration options in the component UI?
talend.input.streaming.maxRecords
talend.input.streaming.maxDurationMs

Reply:
By default, first element checked is localconfiguration then system property. 
We can have a mapping with localconf on UI like this 
```
@Option
@DefaultValue("local_configuration:myfamily.model.key")
private String value;
```
As stated by documentation, some action need to be done to propagate information.

Reply:
yes, in this way we can have this value in UI. 
But as a user, I want to pass the value from the UI to the configuration to stop watch.

Reply:
https://jira.talendforge.org/browse/TCOMP-2226